operator/src/main/resources/scripts/modelInImage.sh

```diff
@@ -1286,21 +1296,25 @@ function prepareMIIServer() {
 
   for file in $(sort_files ${IMG_ARCHIVES_ROOTDIR} "*.zip")
     do
-        # expand the archive domain libraries to the domain lib
-        cd ${DOMAIN_HOME}/lib || return 1
-        ${JAVA_HOME}/bin/jar xf ${IMG_ARCHIVES_ROOTDIR}/${file} ${WLSDEPLOY_DOMAINLIB}
 
-        if [ $? -ne 0 ] ; then
-          trace SEVERE  "Domain Source Type is FromModel, error in extracting domain libs ${IMG_ARCHIVES_ROOTDIR}/${file}"
-          return 1
+        # expand the archive domain libraries to the domain lib, 11 is caution when zip entry doesn't exists
+        cd ${DOMAIN_HOME}/lib || exitOrLoop
+        unzip -jo ${IMG_ARCHIVES_ROOTDIR}/${file} wlsdeploy/domainLibraries/*
+        if [ $? -ne 0 && $? -ne 11 ] ; then
+          trace SEVERE  "Domain Source Type is FromModel, error in extracting domainLibraries
+          ${IMG_ARCHIVES_ROOTDIR}/${file}"
+          exitOrLoop
         fi
 
-        # Flatten the jars to the domain lib root
-
-        if [ -d ${WLSDEPLOY_DOMAINLIB} ] && [ "$(ls -A ${WLSDEPLOY_DOMAINLIB})" ] ; then
-          mv ${WLSDEPLOY_DOMAINLIB}/* .
+        # expand the domain bin, in update case user may only update a file in the domainBin archive, 11 is caution when
+        # zip entry doesn't exists
+        cd ${DOMAIN_HOME}/bin || exitOrLoop
+        unzip -jo ${IMG_ARCHIVES_ROOTDIR}/${file} wlsdeploy/domainBin/*
```

Comment:
For each unzip, please consider capturing stdout & stderr for each unzip to some unique file (or env var), and then 'cat' the capture just after the SEVERE.  This will provide more diagnostic info in the SEVERE. (When the operator parses out a log message and reports it, it appends everything found up until the next log message).

Reply:
There are many lines we simple exit, this kind is rare, I don't think we need every line to be so verbose.